### PR TITLE
Ask pull secret during start

### LINKED ISSUE #18: Handle pullsecret
The extension need to properly request the pullsecret when needed and perform basic validation before setting it.

On startup, when no pull-secret is available, the start will be postponed until the user has provided this; a dialog or input will be shown to collect this.

The user should be abel to provide the pull-secret from the configuration before start.

Basic validation is performed to make sure the pull-secret is a valid json message; a simple `{` and `}` is sufficient as a first increment.

### PR DESCRIPTION
**Pull request by evidolob**
Fixes #18

This PR add check if pullsecret is missing during vm start:

Screenshot 2023-03-21 at 11 37 12

And it add basic validation, it just check that provided value is valid JSON:

Screenshot 2023-03-21 at 11 37 39

>Also this PR provided various refactoring.

### CODE REVIEW

**Comment by evidolob** [comment]
@deboer-tim I add pullsecret validation(copy from crc codebase), and show more descriptive error:
Screenshot 2023-03-21 at 15 16 37

**Comment by gbraad** [comment]
> and show more descriptive error

The ideal is that the pull-secret is automagically retrieved and does not see a copy-paste action. And instead, you will just provide the RH credentials, and see the pull-secret identity in a form as:

```

Pull-secret belongs to:

   g****d@r****t.com

```

**Comment by evidolob** [comment]
@gbraad `"Escape" to cancel... ` is from podman desktop, I cannot change with current API

**Comment by gbraad** [comment]
> I cannot change with current API

Weird text. As an example, vscode does not state this.

**Comment by evidolob** [comment]
vscode shows exactly same text:
Screenshot 2023-03-22 at 09 38 35

File: src/daemon-commander.ts [hunk 1]
```diff
@@ -127,25 +132,40 @@ export class DaemonCommander {
...
-    const { body } = await got.get(url);
+    const { body } = await this.get(url);
     return body;
   }
+
+  private get(url: string): Promise<Response<string>> {
+    return got.get(url, {
+      enableUnixSockets: true,
+      throwHttpErrors: false,
+      retry: {
```
Comment (gbraad) [inline src/daemon-commander.ts:164]: 👍

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
OK, since this is a prompt and not a selection.

I am fine with this as the ideal is that this prompt is not shown at all.

**Pull request merged**